src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -116,15 +122,20 @@ def __init__(self):
                 "start_job_update": self._modify_job_update,
                 "stop_job_update": self._modify_job_update,
                 "cancel_job": self._cancel_job,
-                "retry_job": self._retry_job,
+                "retry_job": self._retry_jobs,
                 "job_logs": self._get_job_logs,
                 "job_logs_latest": self._get_job_logs,
             }
 
     def _verify_job_id(self, req: JobRequest) -> None:
-        if req.job_id is None:
+        if req.job_id is None and req.job_id_list is None:
             self.send_error_message("job_does_not_exist", req)
             raise ValueError(f"Job id required to process {req.request} request")
+        elif req.job_id_list:
+            for job_id in req.job_id_list:
+                if job_id is None:
+                    self.send_error_message("job_does_not_exist", req, {"job_id": job_id})
+                    raise ValueError(f"Job id required to process {req.request} request")
```

Comment:
Add `job_id_list` to `_verify_job_id`

Reply:
You could make life easier for yourself and just filter out any job IDs that are `None` from the id list.

Reply:
From the perspective of the frontend, if there's a `None` instead of a job ID, it means something has gone wrong somewhere. It's not actually helpful to have a message back saying that job ID `None` does not exist, so silently filtering the `None`s out of the `job_id_list` is the most pragmatic approach here.

Reply:
I'm guessing in production-level projects, failing silently may sometimes be preferable to rigor